gateway-js/src/executeQueryPlan.ts

```diff
@@ -551,14 +546,8 @@ function flattenResultsAtPath(value: any, path: ResponsePath): any {
 function downstreamServiceError(
   originalError: GraphQLFormattedError,
   serviceName: string,
-  query: string,
-  variables?: Record<string, any>,
 ) {
-  let {
-    message,
-    extensions,
-    path,
-  } = originalError
+  let { message, extensions, path } = originalError;
```

Comment:
I wonder if we should just remove `path` too.  In a federated request, the `path` here is still not relevant to the request that the client made.  The [specification](https://spec.graphql.org/draft/#sec-Errors.Error-result-format) states:

> If an error can be associated to a particular field in the GraphQL result, it must contain an entry with the key path that details the path of the response field which experienced the error. This allows clients to identify whether a null result is intentional or caused by a runtime error.

I'm not really sure if a client has any reasonable way to make this determination ("allows clients to identitfy ...") given the `path` is mapped to an internal request that the Gateway makes.

This needs some further thought in the federated model, and particularly if the field was inaccessible and merely used to fulfill a different client request.  This relates to https://github.com/apollographql/federation/issues/354.

Any solution here is out of scope for this issue, so I'm tempted to suggest we just remove `path` here too until we can find a principled solution.

My concrete ask is to remove `path` and offer the same suggestion for `variables` and `query` if it's necessary to be preserved.

Reply:
d9e2322